torch/_inductor/autotuner/model.py

```diff
@@ -0,0 +1,649 @@
+        return NN(hidden_dim=[8192, 64], use_norm=False, activation="leaky_relu")
+    else:
+        assert False, "Unknown model type"
+
+
+### feature extraction related
+
+# op_dict needs to be deterministic
+op_dict = {
+    "load": 0,
+    "to_dtype": 1,
+    "add": 2,
+    "reduction": 3,
+    "constant": 4,
+    "div": 5,
+    "store": 6,
+    "sub": 7,
+    "square": 8,
+    "rsqrt": 9,
+    "mul": 10,
+    "tanh": 11,
+    "ne": 12,
+    "where": 13,
+    "indirect_indexing": 14,
+    "log": 15,
+    "neg": 16,
+    "exp": 17,
+    "maximum": 18,
+    "minimum": 19,
+    "index_expr": 20,
+    "ge": 21,
+    "masked": 22,
+    "lt": 23,
+    "and_": 24,
+    "erf": 25,
+    "eq": 26,
+    "le": 27,
+    "gt": 28,
+    "relu": 29,
+    "sqrt": 30,
+    "logical_not": 31,
+    "load_seed": 32,
+    "rand": 33,
+    "abs": 34,
+    "reciprocal": 35,
+    "ceil": 36,
+    "sigmoid": 37,
+    "sin": 38,
+    "cos": 39,
+    "logical_and": 40,
+    "bitwise_and": 41,
+    "randn": 42,
+    "floor": 43,
+    "remainder": 44,
+    "isinf": 45,
+    "logical_or": 46,
+    "expm1": 47,
+    "libdevice_sqrt": 48,
+    "libdevice_log": 49,
+    "truediv": 50,
+    "sign": 51,
+    "randint64": 52,
+    "bitwise_or": 53,
+    "pow": 54,
+    "isnan": 55,
+}
+
+
+class KernelCategory(enumerate):
+    POINTWISE = 0
+    REDUCTION = 1
+    PERSISTENT_REDUCTION = 2
+
+
+def get_kernel_category(src: str) -> KernelCategory:
+    if "@pointwise" in src:
+        return KernelCategory.POINTWISE
+    if "@reduction" in src:
+        return KernelCategory.REDUCTION
+    if "@persistent_reduction" in src:
+        return KernelCategory.PERSISTENT_REDUCTION
+
+
+def get_number_of_loops(src: str) -> int:
+    return src.count("for roffset in range(0, rnumel, RBLOCK):")
+
+
+def parse_list_of_numbers(s: str) -> list:
+    # num1, num2, num3, ...
+    nums = s.strip().split(",")
+    nums = [num.strip() for num in nums]
+    return [int(num) for num in nums]
+
+
+def get_size_hints(src: str) -> list:
+    return parse_list_of_numbers(re.search(r"size_hints=\[([^\]]*)\]", src).group(1))
+
+
+def get_tiling(src: str) -> list:
+    names = ["xnumel", "ynumel", "rnumel"]
+    result = list()
+    for name in names:
+        startpos = src.find(name + " =")
+        if startpos == -1:
+            result.append(1)
+            continue
+        endpos = src.find("\n", startpos)
+        result.append(int(src[startpos + len(name + " = ") : endpos]))
+    return result
+
+
+def pad_tensor():
+    tensor_feature = list()
+    tensor_feature.append(False)  # StarDepOrWeakDep
+    tensor_feature.append(0)  # bytes
+    # we use the lowest 6 dims of the tensor
+    tensor_feature.extend([0] * 6)  # strides,
+    # we use the lowest 6 dims of the tensor
+    tensor_feature.extend([0] * 6)  # size
+    tensor_feature.append(True)  # is_contiguous
+    tensor_feature.append(False)  # is_scalar
+    tensor_feature.append(False)  # is_indirect
+    return tensor_feature
```

Comment:
I am assuming this needs to match other places in the codebase? How do we ensure it stays in sync?